src/Library/demos/Text Input/main.blp

```diff
@@ -0,0 +1,309 @@
+using Gtk 4.0;
+using Adw 1;
+
+Adw.Clamp {
+  Box {
+    orientation: vertical;
+
+    Label {
+      label: "Text Input";
+      margin-top: 10;
+      margin-bottom: 10;
+
+      styles [
+        "title-1",
+      ]
+    }
+
+    Label {
+      label: "Widgets used for reading text input";
+      margin-bottom: 10;
+    }
+
+    Label {
+      label: "Entry";
+      margin-top: 10;
+      halign: start;
+
+      styles [
+        "heading",
+      ]
+    }
+
+    Label {
+      label: "A simple but versatile widget that allows single line text entry and editing.";
+      margin-top: 10;
+      margin-bottom: 10;
+      halign: start;
+
+      styles [
+        "dim-label",
+      ]
+    }
+
+    FlowBox {
+      homogeneous: true;
+      row-spacing: 20;
+
+      Box {
+        orientation: vertical;
+
+        Label {
+          label: "Regular";
+          margin-bottom: 10;
+        }
+
+        Entry entry {
+
+        }
+      }
+
+      Box {
+        orientation: vertical;
+
+        Label {
+          label: "Overwritable";
+          margin-bottom: 10;
+        }
+
+        Entry {
+          overwrite-mode: true;
+          buffer: bind entry.buffer;
+        }
+      }
+
+      Box {
+        orientation: vertical;
+
+        Label {
+          label: "Progress Bar";
+          margin-bottom: 10;
+
+        }
+
+        Entry progress_entry{
+          progress-fraction: 0;
+          buffer: bind entry.buffer;
+        }
+      }
+
+      Box {
+        orientation: vertical;
+
+        Label {
+          label: "Placeholder Text";
+          margin-bottom: 10;
+        }
+
+        Entry {
+          placeholder-text: "Enter text here";
+          buffer: bind entry.buffer;
+        }
+      }
+
+      Box {
+        orientation: vertical;
+
+        Label {
+          label: "Icons";
+          margin-bottom: 10;
+        }
+
+        Entry icon_entry {
+          primary-icon-name: "about-symbolic";
+          primary-icon-activatable: true;
+          primary-icon-tooltip-text: "Click on me!";
+          secondary-icon-name: "bell-symbolic";
+          secondary-icon-tooltip-text: "No click on me!";
+          buffer: bind entry.buffer;
+        }
+      }
+
+      Box {
+        orientation: vertical;
+
+        Label {
+          label: "Visiblity";
+          margin-bottom: 10;
+        }
+
+        Entry visibility_entry {
+          visibility: false;
```

Comment:
I don't see a use case for using this over `PasswordEntry` - which is also safer

You can remove this one too

Reply:
👍🏾